### ISSUE TITLE
Language snippets issue - internal error

### ISSUE THREAD

**Issue by Caseyb87** [spyder-ide/spyder#14426]
### What steps will reproduce the problem?

I have reproduced this several times but it is not entirely consistent (doesn't happen every time).

Split/unsplit a line with a carriage return. Spyder will sometimes run into an internal problem. 

Edit: I think the line needs to be a function definition and you need to split just before an argument name.

### Traceback
```python-traceback
  File "C:\Users\_username_\Anaconda3\lib\site-packages\spyder\plugins\completion\snippets\actor.py", line 92, in handle_msg
    if language_snippets[current_word]:
  File "C:\Users\_username_\Anaconda3\lib\site-packages\spyder\plugins\completion\snippets\trie.py", line 81, in __getitem__
    if len(sequence) == 0:
TypeError: object of type 'NoneType' has no len()
```

## Versions

* Spyder version: 4.2.0 
* Python version: 3.8.3
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  3.0.4 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=4.0                   :  7.19.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
paramiko >=2.4.0                :  2.7.2 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.7.2 (OK)
pygments >=2.0                  :  2.7.3 (OK)
pylint >=1.0                    :  2.6.0 (OK)
pyls >=0.36.1;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.1.1             :  0.2.1 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.1 (OK)
qtconsole >=4.7.7               :  4.7.7 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
rtree >=0.8.3                   :  0.9.4 (OK)
setuptools >=39.0.0             :  51.0.0.post20201207 (OK)
sphinx >=0.6.6                  :  3.2.1 (OK)
spyder_kernels >=1.10.0;<1.11.0 :  1.10.0 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  0.10.4 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  0.29.21 (OK)
matplotlib >=2.0.0              :  3.3.2 (OK)
numpy >=1.7                     :  1.19.2 (OK)
pandas >=1.1.1                  :  1.1.3 (OK)
scipy >=0.17.0                  :  1.5.2 (OK)
sympy >=0.7.3                   :  1.6.2 (OK)
```

**Comment by Caseyb87** [comment]
Looks like there are quite a few ways to run into this bug:
#14428
#14429
#14435

**Comment by bsumlin** [comment]
Per request, here's the column I copy/pasted from Excel into the IDE to get the crash:

300.7762
349.9783
400.7137
449.9049
500.633
549.8205
600.5432
649.7289
700.4497
749.6336
800.3526
849.5347
900.2555
950.9763
1000.157
1050.876
1100.059
1150.778
1199.959

**Comment by steff456** [comment]
@bsumlin can you please upload a gif with the crash? I'm unable to reproduce it

**Comment by steff456** [comment]
It seems this issue is a duplicate of #14183 fixed by #14186, it will be part of our next release. You can follow us in social media if you want to keep track and update as soon as it is available.

Thanks all of reporting this issue and answering so fast!
